Windows build fails

The new License plugin has issues on Windows. My guess is newlines - https://github.com/hierynomus/license-gradle-plugin/issues/68

Lets submit an upstream PR to get this resolved. If we cannot, lets detect OS in build.gradle, if Windows, skip the license checks.

@iBotPeaches 

Thanks for the patch.
Now pass the licence check biut I got aother error during the etsts

> Task :brut.apktool:apktool-cli:compileJava
Note: D:\GIT\Apktool\brut.apktool\apktool-cli\src\main\java\brut\apktool\Main.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :brut.apktool:apktool-lib:test

brut.androlib.UnknownDirectoryTraversalTest > invalidRootFileTest FAILED
    java.lang.AssertionError: Expected exception: brut.common.RootUnknownFileException

95 tests completed, 1 failed, 1 skipped

Thanks.

@iBotPeaches 

The liecense problems seems fixe.
After the update of 12.07.2015 compilation was ok but failed during tests.

> Task :brut.apktool:apktool-cli:compileJava
Note: D:\GIT\Apktool\brut.apktool\apktool-cli\src\main\java\brut\apktool\Main.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :brut.apktool:apktool-lib:test

brut.androlib.UnknownDirectoryTraversalTest > invalidRootFileTest FAILED
    java.lang.AssertionError: Expected exception: brut.common.RootUnknownFileException

95 tests completed, 1 failed, 1 skipped

--- a/brut.apktool/apktool-lib/src/main/java/brut/androlib/res/xml/ResXmlPatcher.java
+++ b/brut.apktool/apktool-lib/src/main/java/brut/androlib/res/xml/ResXmlPatcher.java
@@ -19,8 +19,8 @@ package brut.androlib.res.xml;
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.IOException;
+import java.util.logging.Logger;
 
-import javax.xml.XMLConstants;
 import javax.xml.parsers.DocumentBuilder;
 import javax.xml.parsers.DocumentBuilderFactory;
 import javax.xml.parsers.ParserConfigurationException;
@@ -264,8 +264,12 @@ public final class ResXmlPatcher {
         docFactory.setFeature(FEATURE_DISABLE_DOCTYPE_DECL, true);
         docFactory.setFeature(FEATURE_LOAD_DTD, false);
 
-        docFactory.setAttribute(ACCESS_EXTERNAL_DTD, " ");
-        docFactory.setAttribute(ACCESS_EXTERNAL_SCHEMA, " ");
+        try {
+            docFactory.setAttribute(ACCESS_EXTERNAL_DTD, " ");
+            docFactory.setAttribute(ACCESS_EXTERNAL_SCHEMA, " ");
+        } catch (IllegalArgumentException ex) {
+            LOGGER.warning("JAXP 1.5 Support is required to validate XML");
+        }
 
         DocumentBuilder docBuilder = docFactory.newDocumentBuilder();
         // Not using the parse(File) method on purpose, so that we can control when
@@ -301,4 +305,6 @@ public final class ResXmlPatcher {
     private static final String ACCESS_EXTERNAL_SCHEMA = "http://javax.xml.XMLConstants/property/accessExternalSchema";
     private static final String FEATURE_LOAD_DTD = "http://apache.org/xml/features/nonvalidating/load-external-dtd";
     private static final String FEATURE_DISABLE_DOCTYPE_DECL = "http://apache.org/xml/features/disallow-doctype-decl";
+
+    private static final Logger LOGGER = Logger.getLogger(ResXmlPatcher.class.getName());
 }

